Part_2/PassengerSignals/Passenger.java

```diff
@@ -0,0 +1,102 @@
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.net.Socket;
+import java.io.PrintWriter;
+
+public class Passenger {
+    private PrintWriter ticketWriter;
+    private PrintWriter socketOut;
+    private Socket socket;
+    private BufferedReader socketIn;
+    private OutputFileStream outputStream;
+    private InputFileStream inputStream;
+    private Flight[] receivedFlights;
+
+
+    public Passenger(String serverName, int portNumber) {
+        try {
+            socket = new Socket(serverName, portNumber);
+        } catch (IOException e) {
+            System.err.println(e.getStackTrace());
+        }
+    }
+
+    /**
+     *
+     * @param param The paramater being searched for
+     * @param key The key used for the search
+     * @return The object searched for
+     */
+    public String searchFlights(String param, String key) {
+        String response = "";
+
+        try {
+            socketOut.println("SEARCHFLIGHT_" + param + "_" + key);
+            response = socketIn.readLine();
+
+            if (!response.equals("GOOD")) {
+                inputStream = ObjectInputStream(socket.getInputStream());
+                receivedFlights = (Flight[]) inputStream.readObject();
+                return "GOOD";
+            }
+            else {
+                return "The flight you were looking for could not be found";
```

Comment:
message will be displayed directly to GUI, perhaps make it "No flights found matching criteria"